### ISSUE TITLE
Windows Games with DXVK runs at 25 fps if 32-bit, but 60 fps if 64-bit

### ISSUE THREAD

**Issue by Arucard1983** [felixdoerre/primus_vk#37]
Once I install primus_vk, the last stable NVidia drivers, and Mesa drivers for Intel from Debian 9 backports repository, and playing some games on Steam I notice a curious pattern:

Playing 32-bit Windows games such as Assassin's Creed III (under Steam and Proton 4.2.4), or Assassin's Creed IV (installing Windows Steam Client with Wine-Staging 4.8, plus DXVK 1.2.1), I notice that the games are playable at high settings, but the framerate never goes above 25 fps (20 to 22 fps are the average). 

Installing a 64-bit Windows game that benefit DXVK quickly runs at smooth 60 fps (it may drop to 50, in some situations), for example the Manx TT from Steam.
I have another 64-bit Unity game (Edge of Eternity), but their engine are 30 fps locked.

My hardware specs:
Clevo Laptop with Intel i7-7700HQ and 16 GB of RAM
Intel HD 630 as the display GPU
Nvidia 1050 Ti as the rendered GPU.
All GPU share 3 GB of dedicated memory.

There's some reason to the 32-bit to 64-bit library bridge cut the expected framerate by half ?

**Comment by leonmaxx** [comment]
Possibly that compiler optimizations for 64-bit CPUs not applied when building for 32-bit mode. For example `memcpy` largely benefits from enabled SSE ISA. Try to add following `GCC` options when building primus_vk for 32-bit:  
```
-m32 -march=x86-64 -mtune=generic -mfpmath=sse
```
I use this options when doing my 32-bit builds and I have no problems with performance.

**Comment by Arucard1983** [comment]
It removes the occasional stuttering when compiles the shaders, but the framerate never goes up.
But the most funny thing, is the game are fluid even with 25 fps!

**Comment by Arucard1983** [comment]
I run the game with DXVK_HUD=full to make a better performance benchmark, and displays the real performance of the render GPU (nvidia), which gives between 30 to 50 fps in average, will the display GPU (Intel) always cuts to a smooth 20 to 22 fps.
The tip you provide eliminates some stutter.
Probably is more a bug on Mesa Vulkan drivers on Intel, than a real issue.

**Comment by leonmaxx** [comment]
It's possible that your DE doing some weird stuff with game window.
Try different desktop environment and fullscreen unredirect mode on/off.

**Comment by Arucard1983** [comment]
Finally I manage to fix the issue for Assassin's Creed III and IV Black Flag, it requires to build the 32-bit libraries wit SSE optimizations to fix the shared compiler stutter.
The 20 to 25 fps are a fluke, since the DXVK=full reveal that the real count are 40 to 50 fps on high settings at FullHD.

To make those game enjoyable are necessary to add on Steam shortcut the following system variables:
__GL_THREADED_OPTIMIZATIONS=1 PRIMUS_VK_MULTITHREADING=1 pvkrun %command%

**Comment by felixdoerre** [comment]
Should we add those compile options to the Makefile? i.e:
```
ifeq ($(filter -m32,$(CXXFLAGS),-m32)
  CXXFLAGS += -march=x86-64 -mtune=generic -mfpmath=sse
endif
```
to implicitly document them for other users?

**Comment by Arucard1983** [comment]
Tested a new 32-bit and 64-bit game: Megamagic: Wizard from Neon Age (32) and Witcher 3 (64), both giving high fps using primus_vk with the standard settings. (50 fps or higher at max settings).

The problem that I posted appears to be specific to the Assassin's Creed 3 and 4 games that caps the framerate at 25 fps, and due to unknown causes, they are only smoothly playable with the following hack:
`__GL_THREADED_OPTIMIZATIONS=1 PRIMUS_VK_MULTITHREADING=1 pvkrun %command%`
Running the same games on a tower PC with NVidia 970 GTX, they display locked 60 fps.
I think this is a unrelated  bug  to this project.
